File: Talleres/Taller2/ej2/SubwayCard.java
package Talleres.Taller2.ej2;

public abstract class SubwayCard {
    protected SubwayCentral subwayCentral;

    public SubwayCard(SubwayCentral subwayCentral) {
        this.subwayCentral = subwayCentral;
    }

    protected abstract boolean canRide();

    protected abstract void applyCost();

    public void ride() {
        if(!canRide()) {
            System.out.println("Cannot ride");;
        }
        applyCost();
    }
}

File: POO-Ruby/Guia8/ej6/subway_card.rb
require_relative 'subway_central'
class SubwayCard
  def initialize
    raise 'Subway Card is an abstract class'
  end

  def create(central)
    @central=central
  end

  def ride
    raise 'Undefined method ride'
  end

  def ride?
    raise 'Undefined method ride?'
  end

  def to_s
    "Subway Card [Central: #{@central}]"
  end

  private :ride?, :create

end
